openmdao/core/system.py

```diff
@@ -2020,6 +2020,10 @@ def resolve(to_match, io_types, matches, proms):
                             pmap[name] = gname + name if gname else name
 
             not_found = (set(names).union(renames).union(patterns)) - found
+            if self._var_promotes['any']:
+                if not self._var_abs2meta and isinstance(self, openmdao.core.group.Group):
+                    raise RuntimeError('No inputs or outputs found. Check %s '
```

Comment:
this error message provides no context.  I would recommend, instead of creating a whole new error message, just use the existing one, and if the group happens to be empty, then add the 'check to make sure it's not empty' at the end of the existing message, since that message already provides all of the other relevant information.